WcaOnRails/app/models/registration.rb

```diff
@@ -115,7 +115,8 @@ def paid_entry_fees
   end
 
   def last_payment_date
-    registration_payments.order(created_at: :desc).first&.created_at
+    sorted_payments = registration_payments.sort_by { |p| p.created_at }
+    sorted_payments.first&.created_at
```

Comment:
Why not simply `registration_payments.sort_by(&:created_at).last&.created_at`.
Note that I changed to `last` as before it was `created_at: :desc` but `sort_by` does it in the ascending order, right?